### PR TITLE
Update UPS to Keycloak 1.7.0.Final

### PR DESCRIPTION
**Pull request by abstractj**
See: https://issues.jboss.org/browse/AGPUSH-1528

Requirements:

- Merge https://github.com/aerogear/aerogear-parent/pull/54 first or just clone and run maven install

Some of dependencies like Keycloak connections for Infinispan, are not mandatory, but I decided to include in case of us wanting some clustering features.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -21,7 +21,7 @@
         <groupId>org.jboss.aerogear</groupId>
         <artifactId>aerogear-parent</artifactId>
         <!-- NOTE: if updated for version of UPS dependencies, please update the 'aerogear.bom.version' as well -->
-        <version>0.2.18</version>
+        <version>0.2.18-SNAPSHOT</version>
```
Comment (matzew) [inline pom.xml]: that's actually .19

**Comment by matzew** [comment]
Ok, did a rolling upgrade of 1.1.0 (with data in the DB), and yeah, 1.7.0 works fine:

**Comment by matzew** [comment]
That was on WF-9.

Will do another test of this w/ EAP 6.4.4

**Comment by abstractj** [comment]
@matzew updated! Also, I did a rebase against ag-parent

**Comment by matzew** [comment]
I am getting the following on EAP-6.4.4:
```
09:59:59,361 ERROR [org.jboss.as.controller.management-operation] (DeploymentScanner-threads - 1) JBAS013412: Timeout after [300] seconds waiting for service container stability. Operation will roll back. Step that first updated the service container was 'full-replace-deployment' at address '[]'
10:00:04,366 ERROR [org.jboss.as.controller.management-operation] (DeploymentScanner-threads - 1) JBAS014781: Step handler org.jboss.as.server.deployment.DeploymentHandlerUtil$4@4bbef5d9 for operation {"operation" => "full-replace-deployment","address" => [],"name" => "auth-server.war","content" => [{"path" => "deployments/auth-server.war","relative-to" => "jboss.server.base.dir","archive" => true}],"persistent" => false,"enabled" => true} at address [] failed handling operation rollback -- java.util.concurrent.TimeoutException: java.util.concurrent.TimeoutException
	at org.jboss.as.controller.OperationContextImpl.waitForRemovals(OperationContextImpl.java:275) [jboss-as-controller-7.5.4.Final-redhat-4.jar:7.5.4.Final-redhat-4]
	at org.jboss.as.controller.AbstractOperationContext$Step.handleResult(AbstractOperationContext.java:1155) [jboss-as-controller-7.5.4.Final-redhat-4.jar:7.5.4.Final-redhat-4]
	at org.jboss.as.controller.AbstractOperationContext$Step.finalizeInternal(AbstractOperationContext.java:1108) [jboss-as-controller-7.5.4.Final-redhat-4.jar:7.5.4.Final-redhat-4]
	at org.jboss.as.controller.AbstractOperationContext$Step.finalizeStep(AbstractOperationContext.java:1068) [jboss-as-controller-7.5.4.Final-redhat-4.jar:7.5.4.Final-redhat-4]
	at org.jboss.as.controller.AbstractOperationContext$Step.access$300(AbstractOperationContext.java:1028) [jboss-as-controller-7.5.4.Final-redhat-4.jar:7.5.4.Final-redhat-4]
	at org.jboss.as.controller.AbstractOperationContext.handleContainerStabilityFailure(AbstractOperationContext.java:849) [jboss-as-controller-7.5.4.Final-redhat-4.jar:7.5.4.Final-redhat-4]
	at org.jboss.as.controller.AbstractOperationContext.doCompleteStep(AbstractOperationContext.java:526) [jboss-as-controller-7.5.4.Final-redhat-4.jar:7.5.4.Final-redhat-4]
	at org.jboss.as.controller.AbstractOperationContext.completeStepInternal(AbstractOperationContext.java:338) [jboss-as-controller-7.5.4.Final-redhat-4.jar:7.5.4.Final-redhat-4]
	at org.jboss.as.controller.AbstractOperationContext.executeOperation(AbstractOperationContext.java:314) [jboss-as-controller-7.5.4.Final-redhat-4.jar:7.5.4.Final-redhat-4]
	at org.jboss.as.controller.OperationContextImpl.executeOperation(OperationContextImpl.java:1144) [jboss-as-controller-7.5.4.Final-redhat-4.jar:7.5.4.Final-redhat-4]
	at org.jboss.as.controller.ModelControllerImpl.internalExecute(ModelControllerImpl.java:331) [jboss-as-controller-7.5.4.Final-redhat-4.jar:7.5.4.Final-redhat-4]
	at org.jboss.as.controller.ModelControllerImpl.execute(ModelControllerImpl.java:201) [jboss-as-controller-7.5.4.Final-redhat-4.jar:7.5.4.Final-redhat-4]
	at org.jboss.as.controller.ModelControllerImpl$3$1$1.run(ModelControllerImpl.java:652) [jboss-as-controller-7.5.4.Final-redhat-4.jar:7.5.4.Final-redhat-4]
	at org.jboss.as.controller.ModelControllerImpl$3$1$1.run(ModelControllerImpl.java:646) [jboss-as-controller-7.5.4.Final-redhat-4.jar:7.5.4.Final-redhat-4]
	at java.security.AccessController.doPrivileged(Native Method) [rt.jar:1.8.0_65]
	at org.jboss.as.controller.ModelControllerImpl$3$1.run(ModelControllerImpl.java:646) [jboss-as-controller-7.5.4.Final-redhat-4.jar:7.5.4.Final-redhat-4]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [rt.jar:1.8.0_65]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [rt.jar:1.8.0_65]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [rt.jar:1.8.0_65]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [rt.jar:1.8.0_65]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [rt.jar:1.8.0_65]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [rt.jar:1.8.0_65]
	at java.lang.Thread.run(Thread.java:745) [rt.jar:1.8.0_65]
	at org.jboss.threads.JBossThread.run(JBossThread.java:122) [jboss-threads-2.1.2.Final-redhat-1.jar:2.1.2.Final-redhat-1]

10:00:09,374 ERROR [org.jboss.as.controller.management-operation] (DeploymentScanner-threads - 1) JBAS013413: Timeout after [5000] seconds waiting for service container stability while finalizing an operation. Process must be restarted. Step that first updated the service container was 'full-replace-deployment' at address '[]'
```

Steps to reproduce:
* build current master, and deploy to EAP 6.4.4 ([download](https://www.dropbox.com/s/5jyhj7ib25o324e/jboss-eap-6.4.4.zip?dl=0))
* login and create an application, with a variant
* checkout AGPUSH-1528 branch (I also did a rebase on current upstream master)
* deploy the `auth-server.war` -> OK
* deploy the `ups-as.war` -> above failure

**Comment by matzew** [comment]
Well :-) I tried that now, with the 1.1.0.Final release - and there it worked

**Comment by matzew** [comment]
Not sure what I was :smoking: 

It now went fine from master too.

@abstractj thanks! will get the damn parent release out now, than merge this guy!

:tada:

**Comment by matzew** [comment]
Release vote/thread started here:
http://lists.jboss.org/pipermail/aerogear-dev/2015-December/012178.html

**Comment by matzew** [comment]
landed in https://github.com/aerogear/aerogear-unifiedpush-server/commit/53bf4d0ba2a9fe227bb51548d1d9133454038c65

**Comment by matzew** [comment]
Thanks, @abstractj :tada: :beers:

**Comment by abstractj** [comment]
@matzew

**Pull request closed without merging**